android/src/main/java/io/ionic/portals/reactnative/ReactNativePortalManager.kt

```diff
@@ -0,0 +1,149 @@
+
+        manager.addPortal(portal)
+        return portal
+    }
+
+    fun getPortal(name: String): Portal = manager.getPortal(name)
+
+    fun enableSecureLiveUpdates(keyPath: String) {
+        LiveUpdateManager.secureLiveUpdatePEM = keyPath
+        usesSecureLiveUpdates = true
+    }
+
+    fun registerFromConfigIfAvailable(context: ReactApplicationContext) {
+        reactApplicationContext = context
+        val config = try {
+            context.assets.open("portals.config.json")
+                .bufferedReader()
+                .use(BufferedReader::readText)
+        } catch (e: IOException) {
+            return
+        }
+
+        val configJson = try {
+            JSONObject(config)
+        } catch (e: JSONException) {
+            throw Error("Portals config data is malformed. Aborting.", e)
+        }
+
+        val registrationKey =
+            if (!configJson.isNull("registrationKey")) configJson.getString("registrationKey") else null
+        registrationKey?.let(::register)
+
+        val liveUpdatesKey =
+            if (!configJson.isNull("liveUpdatesKey")) configJson.getString("liveUpdatesKey") else null
+        liveUpdatesKey?.let {
+            LiveUpdateManager.secureLiveUpdatePEM = it
+            usesSecureLiveUpdates = true
+        }
+
+        val portalJsonArray = configJson.getJSONArray("portals")
+
+        for (index in 0 until portalJsonArray.length()) {
+            val portalJson = portalJsonArray.getJSONObject(index)
+            addPortal(portalJson.toReactMap())
+        }
+    }
+}
+
+internal fun JSONObject.toReactMap(): ReadableMap =
+    keys().asSequence().fold(WritableNativeMap()) { map, key ->
+        try {
+            when (val value = get(key)) {
+                is JSONObject -> map.putMap(key, value.toReactMap())
+                is JSONArray -> map.putArray(key, value.toReactArray())
+                is Boolean -> map.putBoolean(key, value)
+                is Int -> map.putInt(key, value)
+                is Double -> map.putDouble(key, value)
+                is String -> map.putString(key, value)
+                null -> map.putNull(key)
+                else -> map.putString(key, value.toString())
+            }
+        } catch (_: JSONException) {
+        }
+
+        return@fold map
+    }
+
+private fun JSONArray.toReactArray(): ReadableArray =
+    (0 until length()).fold(WritableNativeArray()) { array, index ->
+        try {
+            when (val value = get(index)) {
+                is JSONObject -> array.pushMap(value.toReactMap())
+                is JSONArray -> array.pushArray(value.toReactArray())
+                is Boolean -> array.pushBoolean(value)
+                is Int -> array.pushInt(value)
+                is Double -> array.pushDouble(value)
+                is String -> array.pushString(value)
+                null -> array.pushNull()
+                else -> array.pushString(value.toString())
+            }
+        } catch (_: JSONException) {
```

Comment:
Should this throw?

Reply:
Also, this method is for marshaling data to and from Portals to RN